charts/rancher-monitoring/100.1.1+up19.0.3/charts/rancherServiceMetrics/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,55 @@
+{{- $chartLabel := .Values.chartLabel }}
+{{- if eq $chartLabel "" }}
+{{- $chartLabel = (index (lookup "apps/v1" "Deployment" "cattle-system" "rancher").metadata.labels "chart") }}
+{{- end }}
+{{- if ne $chartLabel "" }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+{{ include "kube-prometheus-stack.labels" . | indent 4 }}
+  name: rancher
+  namespace: cattle-system
+spec:
+  endpoints:
+  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+    port: http
+    tlsConfig:
+      caFile: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+      insecureSkipVerify: true
+      serverName: rancher
+  jobLabel: rancher
+  namespaceSelector:
+    matchNames:
+    - cattle-system
```

Comment:
Same ? Although would whole use case break if they changed NS ?

Reply:
Why would rancher not be installed in cattle-system?

Reply:
Rancher does have cattle-system hardcoded, so should be fine.